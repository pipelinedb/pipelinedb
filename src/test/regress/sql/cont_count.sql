CREATE CONTINUOUS VIEW test_count AS SELECT k::text, COUNT(*) FROM stream_cqcount GROUP BY k;

ACTIVATE test_count;

INSERT INTO stream_cqcount (k) VALUES ('x'), ('x'), ('x'), ('x'), ('x'), ('x');
INSERT INTO stream_cqcount (k) VALUES ('x'), ('x'), ('x'), ('x'), ('x'), ('x'), ('y'), ('y'), ('y'), ('y'), ('y'), ('y');

DEACTIVATE test_count;

SELECT * FROM test_count ORDER BY k;

ACTIVATE test_count;

INSERT INTO stream_cqcount (k) VALUES ('x'), ('x'), ('x'), ('x'), ('x'), ('x');
INSERT INTO stream_cqcount (k) VALUES ('x'), ('x'), ('x'), ('x'), ('x'), ('x'), ('y'), ('y'), ('y'), ('y'), ('y'), ('y');

DEACTIVATE test_count;

SELECT * FROM test_count ORDER BY k;

DROP CONTINUOUS VIEW test_count;
