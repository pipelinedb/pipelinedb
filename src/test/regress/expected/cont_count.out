CREATE CONTINUOUS VIEW test_count AS SELECT k::text, COUNT(*) FROM stream_cqcount GROUP BY k;
INSERT INTO stream_cqcount (k) VALUES ('x'), ('x'), ('x'), ('x'), ('x'), ('x');
INSERT INTO stream_cqcount (k) VALUES ('x'), ('x'), ('x'), ('x'), ('x'), ('x'), ('y'), ('y'), ('y'), ('y'), ('y'), ('y');
SELECT * FROM test_count ORDER BY k;
 k | count 
---+-------
 x |    12
 y |     6
(2 rows)

INSERT INTO stream_cqcount (k) VALUES ('x'), ('x'), ('x'), ('x'), ('x'), ('x');
INSERT INTO stream_cqcount (k) VALUES ('x'), ('x'), ('x'), ('x'), ('x'), ('x'), ('y'), ('y'), ('y'), ('y'), ('y'), ('y');
SELECT * FROM test_count ORDER BY k;
 k | count 
---+-------
 x |    24
 y |    12
(2 rows)

DROP CONTINUOUS VIEW test_count;
